lib/facter/simp_puppet_settings.rb

```diff
@@ -3,24 +3,49 @@
 # automatically be able to use the various aspects of the puppet system
 # regardless of the node's platform.
 #
-# Each entry is under a hash entry of its associated section
+# Each entry is under a hash entry of its associated section.
 
 Facter.add(:puppet_settings) do
-  setcode do
-    retval = {}
+   if Object.const_defined?('Puppet') && Puppet.respond_to?(:settings)
+    setcode do
+      retval = {}
+      puppet_settings = Hash[Puppet.settings.map { |k, v| [k, v] }]
 
-    if Object.const_defined?('Puppet') && Puppet.respond_to?(:settings)
-      puppet_settings = Hash[Puppet.settings.map{|k,v| [k,v]}]
+      Puppet.settings.eachsection do |section|
```

Comment:
Maybe we can get this data from [Puppet-as-a-library](https://puppet.com/docs/puppet/5.4/release_notes.html#puppet-as-a-library-pal-tasks-and-scripting) in the future